src/gui/properties/speedplotview.cpp

```diff
@@ -205,69 +214,63 @@ void SpeedPlotView::setGraphEnable(GraphID id, bool enable)
     viewport()->update();
 }
 
-void SpeedPlotView::pushPoint(const SpeedPlotView::PointData &point)
+void SpeedPlotView::pushPoint(const SpeedPlotView::SampleData &point)
 {
-    m_data5Min.push_back(point);
-    m_averager30Min.push(point);
-    m_averager6Hour.push(point);
-    m_averager12Hour.push(point);
-    m_averager24Hour.push(point);
+    for (Averager *averager : {&m_averager5Min, &m_averager30Min
+                                , &m_averager6Hour, &m_averager12Hour
+                                , &m_averager24Hour})
+    {
+        // SPOILER ALERT: conditions are short circuited, make sure push comes first
```

Comment:
So maybe it's better to reformat conditions?
```c++
if (averager->push(point))
{
    if (m_currentAverager == averager)
        viewport()->update();
}
```

Reply:
it looks cool :P will change it :)